rholang/src/test/scala/coop/rchain/rholang/interpreter/MatchTest.scala

```diff
@@ -62,23 +62,33 @@ class VarMatcherSpec extends FlatSpec with Matchers {
     val target: Send =
       Send(Quote(GPrivate("unforgeable")), List(GInt(7), GInt(8)), false, BitSet())
     val pattern: Send =
-      Send(ChanVar(FreeVar(0)), List(EVar(wc), GInt(8)), false, BitSet())
-    val result = spatialMatch(target, pattern).runS(emptyMap)
-    result should be(Some(Map[Int, Par](0 -> GPrivate("unforgeable"))))
+      Send(ChanVar(FreeVar(0)), List(EVar(wc), GInt(8)), false, BitSet(), true)
+    val expectedResult = Some(Map[Int, Par](0 -> GPrivate("unforgeable")))
+    val result         = spatialMatch(target, pattern).runS(emptyMap)
+    result should be(expectedResult)
+    val targetPar: Par  = target
+    val patternPar: Par = pattern
+    val parResult       = spatialMatch(targetPar, patternPar).runS(emptyMap)
```

Comment:
Are you running `spatialMatch` on the same `target` and `pattern` but cased as different types (`Par` and `Send`) ? Is it bug or a feature? as I understand the implicit conversion is going to kick in here. Maybe we should be more rigid in our type safety?

Reply:
Yes, they're different types.
Yes, this is a feature.
Writing out par expressions without the implicits is a pain.
If you want, you can pull some of the implicits out and put them in a testing only include, because that's where we construct the vast majority of Pars by hand.

Reply:
I understand that implicit conversion is a feature, what I meant to ask with my question was whether we want to have to care that every expression is a `Par` hence it can be implicitly converted to it and then the function on `Par` will be used.